src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeFeature.NonNativeAot.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.CompilerServices
+{
+    public static partial class RuntimeFeature
+    {
+        public static bool IsDynamicCodeSupported { get; } =
+            AppContext.TryGetSwitch("System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported", out bool isDynamicCodeSupported) ? isDynamicCodeSupported : true;
+
+        public static bool IsDynamicCodeCompiled => IsDynamicCodeSupported;
```

Comment:
Should this be [Intrinsic] as it was in https://github.com/dotnet/runtime/pull/80246/files#diff-9a100e91637eec17cfb3d62048bbddeff73e3accf6938596232978814144ba7aL17?

Reply:
I waffled on this, and don’t really understand the value of it being Intrinsic on mono. I didn’t see any code in Mono looking for that attribute.

And now it isn’t Intrinsic all the time. It is only Intrinsic on fullaot and IsDynamicCodeCompiled is only intrinsic on the interpreter. Most of the time, the C# code is used now.

Reply:
@BrzVlad @vargaz @lambdageek - any thoughts here? How does Mono handle "partial" Intrinsic methods?

Reply:
Mono doesn't really check for that attribute, since its not used consistently, it checks for methods by name/signature. [Intrinsic] is still useful to mark methods which the runtime might special case.